src/core.js

```diff
@@ -2158,10 +2168,15 @@ imgix.fluid = function(elem) {
 	var fluidElements;
 	if (elem && !imgix.helpers.isFluidSet(elem)) {
 		fluidElements = Array.isArray(elem) ? elem : [elem];
-	} else {
+  } else {
 		var cls = '' + options.fluidClass;
 		cls = cls.slice(0, 1) === '.' ? cls : ('.' + cls);
-		fluidElements = document.querySelectorAll(cls);
+    if (node && imgix.isImageElement(node)) {
```

Comment:
This shortcut path is smart, but fails to take into account a couple of things.

* `imgix.isImageElement` only checks for `<img>` tags. We should also check here whether the node has a background image. `imgix.hasImage()` will do the trick, but...
* What if this node has a background image and also contains image-y children? This is where the intended behavior gets a little fuzzy. Intuitively, I think both the container and its children should be made fluid, but I'm curious to hear your thoughts.